avail/src/main/kotlin/avail/persistence/cache/Repository.kt

```diff
@@ -154,7 +154,7 @@ class Repository constructor(
 	 * @author Mark van Gulik &lt;mark@availlang.org&gt;
 	 */
 	private object IndexedRepositoryBuilder : IndexedFileBuilder(
-		"Avail compiled module repository V17")
+		"Avail compiled module repository V18")
```

Comment:
How do we decide that we should bump this version? Is it because this change should invalidate the index, or is this just a description that does not impact the Avail build or runtime?

Reply:
Yes, I bumped it because it changed StylingRecord incompatibly, and that's one of the structures that gets written per-module in the repository files.  If we ever make a breaking change and forget to bump that version (it happens), you can just clear your `~.avail/repositories` directory.  The next build of Avail code will simply do a compile from the source, repopulating those files.  That string occurs in the initial few bytes of a `.repo` file, since repositories use a general compressed, transactional, indexed file format that I invented and have reused for other purposes.

Reply:
To clarify, I bumped it because the read/write serialized format of `StylingRecord` changed incompatibly.  The in-memory representation doesn't affect it.

Reply:
As a subtle point, I also changed `StylingRecord.kt:81` to sort the style names.  That part was *not* a breaking change, because the reader doesn't require them to occur in sorted order.  It just makes the zlib compression used by .repo records slightly more effective.